runtime/src/tasks/qthreads/tasks-qthreads.c

```diff
@@ -682,6 +682,44 @@ void chpl_task_init(void)
 
     if (verbosity >= 2) { chpl_qt_setenv("INFO", "1", 0); }
 
+    // Explicitly bind shepherd threads to cores/PUs when using the
+    // binders topology.
+    char* topo = getenv("CHPL_QTHREAD_TOPOLOGY");
+    if ((topo != NULL) && !strcmp(topo, "binders")) {
+        int *cpus = NULL;
+        int numCpus;
+        chpl_bool physical;
+        char *unit = chpl_qt_getenv_str("WORKER_UNIT");
+        if ((unit != NULL) && !strcmp(unit, "pu")) {
+            physical = false;
+            numCpus = chpl_topo_getNumCPUsLogical(true);
+        } else {
+            physical = true;
+            numCpus = chpl_topo_getNumCPUsPhysical(true);
+        }
+        int numShepherds = (int) chpl_qt_getenv_num("NUM_SHEPHERDS", numCpus);
+        if (numShepherds < numCpus) {
+            numCpus = numShepherds;
+        }
+        cpus = (int *) chpl_malloc(numCpus * sizeof(*cpus));
+        numCpus = chpl_topo_getCPUs(physical, cpus, numCpus);
+        int bufSize = numCpus * 5;
```

Comment:
Why `* 5`?

Reply:
That gives us 4 digits of CPU ID and 1 for the ':' separator. I figure 10K cores is more than enough.

Reply:
Can you add a comment to that, or size it based on the number of digits? Maybe something like:

```c
int numDigits = ceil(log10(numCpus));
int cpuBindStrSize = numCpus * (numDigits + 1);
char *cpuBindStr = chpl_malloc(cpuBindStrSize);
```

This can be done in a followup PR too. I don't want to get too caught up in nitpicks and code improvements while we're trying to make progress, but also don't want to lose track of improvement we could make down the road.

Reply:
I'll make the change.